modules/platforms/cpp/core-test/src/cluster_group_test.cpp

```diff
@@ -195,6 +282,30 @@ BOOST_AUTO_TEST_CASE(IgniteForHost)
     BOOST_REQUIRE(group2.GetNodes().size() == 4);
 }
 
+BOOST_AUTO_TEST_CASE(IgniteForHostName)
+{
+    IgniteCluster cluster = server1.GetCluster();
+
+    BOOST_REQUIRE(cluster.IsActive());
+
+    std::string hostName = "someHostName";
+
+    BOOST_CHECK_THROW(cluster.AsClusterGroup().ForHost(hostName), IgniteError);
```

Comment:
I did not find a case when, the cluster node host name attribute is not empty.
The only place where this function is tested is ClusterGroupHostsSelfTest on java side.
But, the host names should be passed to IgniteConfiguration::setDiscoverySpi() before call the ForHost().
If you know some other way to set the host name for cluster node, it would be great.

Reply:
You can configure it using XML config.

Reply:
Try this property:
```
<bean id="grid.cfg" class="org.apache.ignite.configuration.IgniteConfiguration">
    ...
    <property name="localHost" value="192.168.1.42"/> <--- your IP here
    ...
</bean>
```

By the way, in Java this method does not throw if the hostname is unknown. It returns empty group instead. Why does it throw in C++?

Reply:
The host name declared in the xml file, like you explained above, cannot be found in the ClusterNode host names list.